### DISCUSSION TITLE
Should fpm attempt to fix known issues in compilers

### DISCUSSION THREAD

**Discussion by awvwgk** [fortran-lang/fpm#447] [General]
Related to GFortran not using the include paths of the C compiler (`/usr/include`, `/usr/local/include`, ...) to search for module files, but packages installing under those directories assuming modules will be found.

This bug was reported 2008 for GFortran 4.3.0: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=35707

**Comment by certik** [comment]
That particular bug is a bit unclear if it is a bug: Fortran should probably have a place to store mod files and given that they are all compiler version dependent, it should probably be prefixed by the compiler+version. That is something we can do in fpm.

**Reply by awvwgk** [reply]
Well, it is partly netcdf's fault for installing module files in the global include directory and GFortran coping with this by removing a feature to avoid the resulting trouble IMO, which in turn causes more trouble elsewhere.

**Reply by certik** [reply]
I ran into this myself, I even asked at the gfortran's mailinglist long time ago (it might have triggered this bug report, I can't remember). The root of the problem is that Fortran's mod files are more like Python `.pyc` files than like C `.h` files. We do not store Python's `.pyc` files in `$PREFIX/include`, because they are Python version dependent and binary.

But as part of our `fpm` effort, we can develop a good standard / solution to this I think.

**Reply by awvwgk** [reply]
My issue with this is that the GFortran behaviour removes completely viable methods (pkg-config, `CPATH`, environment modules) for solving this problem from the table. Intel Fortran actually relies on `CPATH` to point its compiler to the correct module files.

**Comment by milancurcic** [comment]
I suggest a small shift in perspective. Rather than asking whether fpm should fix issues in compilers, we should ask what does fpm need to do to provide the best possible experience for most users. If that means implementing a workaround for specific compiler design choices--like gfortran's don't look in system paths or use CPATH--then yes. Otherwise, no.

**Comment by ivan-pi** [comment]
Should `fpm` (and the larger fortran-lang community) introduce it's own choice of system folder for Fortran include files?

On the other hand installing `.mod` files should be discouraged.

**Reply by certik** [reply]
I don't know currently if we should, I would first get more experience with `fpm` itself and how people use it and how it interacts with external packages.

However, if we decide as a community that we should down the road, I think we have enough momentum that we can do it. We can get together (video calls), invite GFortran developers, design it, discuss at our Discourse, advertise on Twitter, iterate, etc.
